src/app/(private)/emergency-evacuation-applications/page.tsx

```diff
@@ -0,0 +1,100 @@
+'use client'
+
+import { useEffect, useState } from 'react'
+import PrivateRoute from '@/app/hocs/isAuth'
+import { useTranslation } from 'react-i18next'
+import { useToast } from '@/components/ui/use-toast'
+import { Toaster } from '@/components/ui/toaster'
+import { useDataTable } from '@/app/hocs/useDataTable'
+import { DataTable, DataTableToolbar } from '@/components/dataTable'
+import { useSearchParams } from 'next/navigation'
+import filterFields from '@/modules/emergencyEvacuationApplications/constants/filterFields'
+import { postEmergencyEvacuationApplications } from '@/modules/emergencyEvacuationApplications/service'
+import { columns } from '@/modules/emergencyEvacuationApplications/components/columns'
+import { Permission } from '@/constants/permissions'
+import { searchParamsSchema } from '@/modules/emergencyEvacuationApplications/constants/searchParamsSchema'
+import { EmergencyEvacuationApplications } from '@/modules/emergencyEvacuationApplications/constants/types'
+import FilterInput from '@/components/ui/filterInput'
+
+const Page = () => {
+  const searchParams = useSearchParams()
+  const search = searchParamsSchema.parse(
+    Object.fromEntries(searchParams.entries()),
+  )
+
+  const { t } = useTranslation()
+  const { toast } = useToast()
+  const [data, setData] = useState<EmergencyEvacuationApplications>({
+    content: [],
+    totalPageCount: 0,
+  })
+  const [isLoading, setIsLoading] = useState<boolean>(false)
+  const [error, setError] = useState<string | null>(null)
+
+  useEffect(() => {
+    setIsLoading(true)
+    postEmergencyEvacuationApplications({
+      page: search.page,
+      per_page: search.per_page,
+      sort: search.sort,
+      status: search.status,
+      referenceNumber: search.referenceNumber,
+      seatingCount: Number(search.seatingCount),
+      sourceCity: search.sourceCity,
+      sourceDistrict: search.sourceDistrict,
+      targetCity: search.targetCity,
+      targetDistrict: search.targetDistrict,
+      isInPerson: search.isInPerson,
+    })
+      .then((responseData) => {
+        setData(responseData.data.response)
+      })
+      .catch((error) => {
+        setError(error.message)
+        toast({
+          title: t('error'),
+          description: t('defaultError'),
+          variant: 'destructive',
+        })
+      })
+      .finally(() => setIsLoading(false))
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
ESLint için uyarı atlama commitlerini kullanmak pek de iyi bir yöntem olmayabilir, çünkü bu, potansiyel hataları gözden kaçırmamıza neden olabilir ve kodun beklenmedik davranışlar sergilemesine yol açabilir. Bağımlılıkları tek tek eklemek uzun gelebilir fakat , ESLint disabled satırlarını kullanmamak daha kesin ve öngörülebilir test ve çözümler üretmemizi sağlar.